datafusion/common/src/test_util.rs

```diff
@@ -52,6 +52,22 @@ macro_rules! assert_batches_eq {
     };
 }
 
+#[macro_export]
+macro_rules! assert_snapshot {
+    ($CHUNKS: expr) => {
+        let formatted = $crate::arrow::util::pretty::pretty_format_batches_with_options(
+            $CHUNKS,
+            &$crate::format::DEFAULT_FORMAT_OPTIONS,
+        )
+        .unwrap()
+        .to_string();
+
+        let actual_lines: Vec<&str> = formatted.trim().lines().collect();
+
+        insta::assert_yaml_snapshot!(actual_lines);
+    };
+}
+
```

Comment:
Why do we use this for tests?

Reply:
test with `insta`. Because changing them manually is painful. I did this in `datafusion-example`, since they are not `test`.